pkg/service/main.go

```diff
@@ -90,17 +90,13 @@ func Main(serviceName string, serviceVersion string, proto interface{}, ctx cont
 }
 
 func setServiceName(name string, profile string) string {
-	envValue := os.Getenv(common.EnvServiceName)
+	envValue := os.Getenv(common.EnvInstanceName)
 	if len(envValue) > 0 {
-		serviceNameOverride = envValue
+		instanceName = envValue
 	}
 
-	if len(serviceNameOverride) > 0 {
-		name = serviceNameOverride
-	}
-
-	if len(profile) > 0 {
-		name = name + ";" + profile
+	if len(instanceName) > 0 {
+		name = instanceName + "_" + name
```

Comment:
the instanceName is suffix, not prefix, so it should be `name + "_" + instanceName`